src/values.test.ts

```diff
@@ -9,3 +9,41 @@ describe('Test for values as data first', () => {
     expect(values({ a: 'x', b: 'y', c: 'z' })).toEqual(['x', 'y', 'z']);
   });
 });
+
+describe('typing', () => {
+  it('should correctly types indexed types', () => {
+    // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+    expectTypeOf(values<{ [index: string]: string }>({ a: 'b' })).toEqualTypeOf<
+      Array<string>
+    >();
+  });
+
+  it('should correctly type functions', () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    expectTypeOf(values(function namedFunction() {})).toEqualTypeOf<
+      Array<never>
+    >();
+  });
+
+  it('should correctly type arrays', () => {
+    expectTypeOf(values([1, 2, 3])).toEqualTypeOf<Array<number>>();
+  });
+
+  it('should correctly type const arrays', () => {
+    expectTypeOf(values([1, 2, 3] as const)).toEqualTypeOf<Array<1 | 2 | 3>>();
+  });
+
+  it('should correctly type objects', () => {
+    expectTypeOf(values({ a: true })).toEqualTypeOf<Array<boolean>>();
+  });
+
+  it('should correctly type Records', () => {
+    expectTypeOf(values({ a: true })).toEqualTypeOf<Array<boolean>>();
+  });
```

Comment:
This is the same test, did you mean to use a literal generic value here?
```suggestion
  it('should correctly type Records', () => {
    expectTypeOf(values<Record<string, boolean>>({ a: true })).toEqualTypeOf<Array<boolean>>();
  });
```

Reply:
I did indeed